src/Adapter/Order/OrderProductQuantityUpdater.php

```diff
@@ -223,8 +224,26 @@ private function applyOtherProductUpdates(
                     $orderInvoice,
                     false
                 );
+            } else {
+                foreach ($cart->getProducts() as $product) {
+                    if ((int) $product['id_product'] === $updatedProduct->getProductId()->getValue()
+                        && (int) $product['id_product_attribute'] === $updatedCombinationId) {
+                        $productsToAdd[] = $product;
+                        break;
+                    }
+                }
             }
         }
+        if (count($productsToAdd) > 0) {
```

Comment:
maybe adding a comment here about why this is needed would be helpful for future developers stumbling on this code 👍

Reply:
I thought that comments [here](https://github.com/PrestaShop/PrestaShop/pull/23138/files#diff-e7ebd437c66da6dcc33d47bea4cf3e4d5c528b1380f92f760b7a117d13ea6952R239-R240) were enough

Reply:
It's not in the same file  (am I missing something?)

But my comment is not blocking the PR anyway, just a suggestion